### ISSUE TITLE
Add immediate visual effect when server side is rendering to mitigate feeling of latency

### ISSUE THREAD

**Issue by wanton7** [aspnet/Blazor#1241]
I would like to have support for adding a immediate visual effect to client side when server side is rendering. It would make UI feel more responsive in high latency situations.

Like adding class to a button when it's clicked and remove that class when server responds.
Having way to add spinners to some div when specific event is triggered would be awesome too. Let's say you click a button to update a grid and you could have spinner top of that grid until server responds.

**Comment by simonziegler** [comment]
Hi @wanton7, I wanted the same thing and got a result in a couple of ways:

- I implemented the GrayOutZone stuff that @SteveSandersonMS applies in the Flight Finder example.
- I have some elements that look like `<span class="@MySpecialClass" …` where 'MySpecialClass' is a variable that I change before submitting a server request and reset back once the server responds and I have completed Client side calculation.
- Lastly I have an in place spinner at the top of my page. It is always spinning. I use some JavaScript interop to change its display property to block when I submit a server request and back to none when the server (and client calculation) is done.


Hope that helps.

**Comment by wanton7** [comment]
@simonziegler I'm talking about server side rendering. I mean this new stuff that came out in Blazor 0.5 that renders everything in server side and client side is just a very thin client.

**Comment by simonziegler** [comment]
@wanton7 - got it and my bad. I intend to experiment with this too. I hope and suspect that the same things may apply, and will report back here when or if I find out. I would be very interested to know your investigative results.

**Comment by shadowmint** [comment]
> We're hoping that people will be able to deploy their apps in such a way that latency is on the order of < 200ms (preferably < 50ms)

@SteveSandersonMS Could you possibly comment on how you imagine this is going to be possible, given the average latency for mobile devices is 50ms - 200ms across various providers (and thats in high speed networks), regardless of the deployment target?

I can't see how this is going to work.

**Comment by danroth27** [comment]
@shadowmint you are right that the inherent latency in the server side Blazor model may not make it suitable for all apps or environments. I would think low latency should be achievable in more controlled environments like apps deployed to local data centers and accessed over high bandwidth networks. Apps that don't require a high degree of interactivity should be fine in higher latency environments.
